Update docker registry to `cr.l5d.io`

Pothulapati opened the pull request:
This PR updates the docker registry to be `cr.l5d.io` to make
it consistent with that of the main docker images.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

alpeb reviewed (commented):
Can you please rebase? It seems this is a little behind from `main`...

alpeb commented:
My bad, branch is in sync, I was confused by changes we had done in the release workflow that were not synced against the integration tests workflow.

alpeb commented on .github/workflows/release.yml:
```diff
@@ -7,7 +7,7 @@ permissions:
 env:
   GH_ANNOTATION: true
-  DOCKER_REGISTRY: ghcr.io/linkerd
+  DOCKER_REGISTRY: cr.l5d.io/linkerd
```
The `cr.l5d.io` domain works when pulling images, but I'm not sure if it'll work when pushing, because we still need to tag the image as `ghcr.io/linkerd/linkerd-smi:tag`. That's how it's setup in the release workflow in the linkerd2 repo...

alpeb commented on .github/workflows/integration_tests.yml:
```diff
@@ -27,7 +27,7 @@ jobs:
         echo "TAG=$(CI_FORCE_CLEAN=1 bin/root-tag)" >> $GITHUB_ENV
 
-        echo "DOCKER_REGISTRY=ghcr.io/linkerd" >> $GITHUB_ENV
+        echo "DOCKER_REGISTRY=cr.l5d.io/linkerd" >> $GITHUB_ENV
```
This is the default in `bin/docker-build`, so we can get rid of this line.

Pothulapati commented:
@alpeb, To remove confusing around docker image tags in workflows. I updated it to be the following in the latest commit.

> Everything uses `cr.l5d.io, which is the default in `./bin/docker-build`
and updated *only* the `docker_build` job in `release` workflow
to use `ghcr.io` registry so that it can push images.
